include/boost/math/tools/polynomial.hpp

```diff
@@ -644,6 +645,30 @@ bool even(polynomial<T> const &a)
     return !odd(a);
 }
 
+template <class T>
+polynomial<T> pow(polynomial<T> base, int exp)
+{
+    if (exp < 0)
+        return policies::raise_domain_error(
+                "boost::math::tools::pow<%1%>",
+                "Negative powers are not supported for polynomials.",
+                base, policies::policy<>());
+        // if the policy is ignore_error or errno_on_error, raise_domain_error
+        // will return std::numeric_limits<polynomial<T>>::quiet_NaN(), which
+        // defaults to polynomial<T>(), which is the zero polynomial
+    polynomial<T> result(T(1));
+    if (exp & 1)
+        result = base;
+    /* "Exponentiation by squaring" */
+    while (exp >>= 1)
+    {
+        base *= base;
+        if (exp & 1)
+            result *= base;
+    }
+    return result;
```

Comment:
For an int, I would be surprised if testing `exp & 1` vs `n % 2 == 1` were actually different performance-wise  (I'd expect the compiler to emit the same assembly.) But that version uses `y = x * y` and `x = x * x` instead of `*=`, creating lots of extra temporaries .